modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts

```diff
@@ -0,0 +1,264 @@
+
+            return;
+        }
+
+        await prismaBulkExecuteOperations(
+            [
+                prisma.prismaUser.createMany({
+                    data: userAddresses.map((userAddress) => ({ address: userAddress })),
+                    skipDuplicates: true,
+                }),
+                ...amountUpdates.map((update) => {
+                    const pool = pools.find((pool) => pool.staking?.id === update.farmId);
+                    const farm = farms.find((farm) => farm.id === update.farmId);
+
+                    return prisma.prismaUserStakedBalance.upsert({
+                        where: { id: `reliquary-${update.farmId}-${update.userAddress}` },
+                        update: {
+                            balance: update.amount,
+                            balanceNum: parseFloat(update.amount),
+                        },
+                        create: {
+                            id: `${update.farmId}-${update.userAddress}`,
+                            balance: update.amount,
+                            balanceNum: parseFloat(update.amount),
+                            userAddress: update.userAddress,
+                            poolId: pool!.id,
+                            tokenAddress: farm!.poolTokenAddress,
+                            stakingId: update.farmId,
+                        },
+                    });
+                }),
+                prisma.prismaUserBalanceSyncStatus.update({
+                    where: { type: 'RELIQUARY' },
+                    data: { blockNumber: endBlock },
+                }),
+            ],
+            true,
+        );
+    }
+
+    public async initStakedBalances(): Promise<void> {
+        const { block } = await reliquaryService.getMetadata();
+        console.log('initStakedReliquaryBalances: loading subgraph relics...');
+        const relics = await reliquaryService.getAllRelics({});
+        console.log('initStakedReliquaryBalances: finished loading subgraph relics...');
+        console.log('initStakedReliquaryBalances: loading pools...');
+        const pools = await prisma.prismaPool.findMany({ select: { id: true, address: true } });
+        console.log('initStakedReliquaryBalances: finished loading pools...');
+        const userAddresses = _.uniq(relics.map((relic) => relic.userAddress.toLowerCase()));
+
+        console.log('initStakedReliquaryBalances: performing db operations...');
+
+        await prismaBulkExecuteOperations(
+            [
+                prisma.prismaUser.createMany({
+                    data: userAddresses.map((userAddress) => ({ address: userAddress })),
+                    skipDuplicates: true,
+                }),
+                prisma.prismaUserStakedBalance.deleteMany({}),
+                prisma.prismaUserStakedBalance.createMany({
+                    data: relics.map((relic) => {
+                        const pool = pools.find((pool) => isSameAddress(pool.address, relic.pool.poolTokenAddress));
```

Comment:
In theory, a pool could have a farm both on masterchef and on reliquary. I think that causes a problem down the line. 

Any way we can make sure this doesn't happen?

Reply:
mh from what i see the resolver would just pass down both staking options, farm (masterxhef) & reliquary in the `GqlPoolStaking` type. But then it should never be that both have emissions and i think this can only really happen when we migrate the other pools beside fbeets since fbeets gets a new pool anyways. So we'd have to clean up the database when we do the migration but think there is quitet some more stuff to think about when we do this